lib/portlayer/exec/exec.go

```diff
@@ -113,11 +113,14 @@ func Init(ctx context.Context, sess *session.Session, source extraconfig.DataSou
 func eventCallback(ie events.Event) {
 	// grab the container from the cache
 	container := Containers.Container(ie.Reference())
+
 	if container != nil {
+		container.m.Lock()
```

Comment:
just take a look at eventCallback

```
newState := eventedState(ie.String(), container.state)
if newState != container.state {
....
    container.swapState(newState)
}
```

I would like to keep those three operations protected, so when a new state is set it is guaranteed that previous state hasn't changed.